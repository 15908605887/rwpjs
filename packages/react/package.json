{
  "name": "react",
  "version": "0.0.1",
  "description": "",
  "main": "src/index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "zhangj",
  "license": "MIT",
  "dependencies": {
    "jest": "^26.0.1",
    "lodash": "^4.17.15",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "webpack": "^4.43.0"
  },
  "devDependencies": {
    "@babel/core": "^7.9.6",
    "@babel/plugin-proposal-class-properties": "^7.8.3",
    "@babel/plugin-proposal-object-rest-spread": "^7.9.6",
    "@babel/plugin-transform-typescript": "^7.9.6",
    "@babel/preset-env": "^7.9.6",
    "@babel/preset-react": "^7.9.4",
    "@babel/preset-typescript": "^7.9.0",
    "babel-loader": "^8.1.0",
    "html-webpack-plugin": "^4.3.0",
    "webpack-dev-server": "^3.11.0"
  }
}
