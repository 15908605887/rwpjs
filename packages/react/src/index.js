
exports.default = function (){
    // 预设webpack属性
    const webpackConfig = {
        mode: 'development',
        resolve: {
            extensions: ['.ts', '.tsx', '.js']
        },
        module: {
            rules: [
                {
                    loader: "babel-loader",
                    test: /\.(ts|js)x?$/,
                    exclude: /node_modules/,
                    options: {
                        presets: ['@babel/preset-env', '@babel/preset-react', '@babel/preset-typescript'],
                        plugins: ['@babel/proposal-class-properties', '@babel/proposal-object-rest-spread']
                    },
                }
            ]
        },
        plugins: [
            new HtmlWebpackPlugin()
        ]
    }
    const compiler = Webpack(webpackConfig)
    const server = new WebpackDevServer(compiler, {
        hot: true
    });
    server.listen(8000)
}

