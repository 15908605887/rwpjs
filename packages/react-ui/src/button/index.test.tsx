import React, { useState } from 'react';
import { render, fireEvent, act } from '@testing-library/react'
import { Button } from '../index';

const TestButton = () => {
    const [title, setTitle] = useState<string>('test click')
    return (
        <Button
            onClick={() => {
                setTitle('click')
            }}
        >
            {title}
        </Button>
    )
}

const AsyncTestButton = () => {
    const [title, setTitle] = useState<string>('test click')
    return (
        <Button
            onClick={async () => {
                setTitle('click')
            }}
        >
            {title}
        </Button>
    )
}

const NoAutoLoadingTestButton = () => {
    const [title, setTitle] = useState<string>('test click')
    return (
        <Button
            autoLoading={false}
            onClick={async () => {
                setTitle('click')
            }}
        >
            {title}
        </Button>
    )
}

const NoClickButton = () => (
    <Button
        autoLoading={false}
    >
    </Button>
)

test('test button click .', async () => {
    const { container, findByText } = render(<TestButton />)
    const button = await findByText('test click')
    fireEvent.click(button)
    expect(button.textContent).toBe('click')
    expect(container).toBeInstanceOf(Element)
    expect(container).toMatchSnapshot()
})

test('test button async click .', async () => {
    const { container, findByText } = render(<AsyncTestButton />)
    const button = await findByText('test click')
    await act(async () => {
        await fireEvent.click(button)
    })
    expect(button.textContent).toBe('click')
    expect(container).toBeInstanceOf(Element)
    expect(container).toMatchSnapshot()
})

test('test button no auto loading', async () => {
    const { container, findByText } = render(<NoAutoLoadingTestButton />)
    const button = await findByText('test click')
    await act(async () => {
        await fireEvent.click(button)
    })
    expect(button.textContent).toBe('click')
    expect(container).toBeInstanceOf(Element)
    expect(container).toMatchSnapshot()
});

test('test button no click', async () => {
    const { container } = render(<NoClickButton />)
    expect(container).toMatchSnapshot()
});
