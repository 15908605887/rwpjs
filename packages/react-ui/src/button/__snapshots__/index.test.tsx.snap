// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test button async click . 1`] = `
<div>
  <button
    ant-click-animating-without-extra-node="true"
    class="ant-btn"
    type="button"
  >
    <span>
      click
    </span>
  </button>
</div>
`;

exports[`test button click . 1`] = `
<div>
  <button
    ant-click-animating-without-extra-node="false"
    class="ant-btn"
    type="button"
  >
    <span>
      click
    </span>
  </button>
</div>
`;

exports[`test button no auto loading 1`] = `
<div>
  <button
    ant-click-animating-without-extra-node="true"
    class="ant-btn"
    type="button"
  >
    <span>
      click
    </span>
  </button>
</div>
`;

exports[`test button no click 1`] = `
<div>
  <button
    class="ant-btn"
    type="button"
  />
</div>
`;
